src/renderer/features/network/NetworkSelector/lib/network-selector-utils.ts

```diff
@@ -0,0 +1,23 @@
+import { ExtendedChain } from '@entities/network';
+import { ConnectionType } from '@shared/core';
+import { ConnectionList } from './types';
+
+export const networkSelectorUtils = {
+  getConnectionsList,
+};
+
+function getConnectionsList(networkItem: ExtendedChain): ConnectionList[] {
+  const { connection, nodes } = networkItem;
+  const { canUseLightClient, customNodes } = connection;
+
+  const actionNodes = [
+    ...(canUseLightClient ? [{ type: ConnectionType.LIGHT_CLIENT }] : []),
+    { type: ConnectionType.AUTO_BALANCE },
+    { type: ConnectionType.DISABLED },
+  ];
+
+  const customs = (customNodes || []).map((node) => ({ node, type: ConnectionType.RPC_NODE, isCustom: true }));
+  const nodesWithType = nodes.map((node) => ({ node, type: ConnectionType.RPC_NODE, isCustom: false }));
```

Comment:
Even though I liked the initial variant the most, the suggested variant is a bit more readable 😉
```suggestion
  const customs = (customNodes || []).map((node) => ({
    node,
    type: ConnectionType.RPC_NODE,
    isCustom: true,
  }));
  const nodesWithType = nodes.map((node) => ({
    node,
    type: ConnectionType.RPC_NODE,
    isCustom: false,
  }));
```